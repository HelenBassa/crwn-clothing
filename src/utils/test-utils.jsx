import { render } from '@testing-library/react';
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import { store } from "../store/store";


afterAll(() => jest.clearAllMocks());

const AllTheProviders = ({ children }) => {
  return (
    <Provider store={store}>
      <Router>
        {children}
      </Router>
    </Provider>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export * from '@testing-library/jest-dom';
export * from '@testing-library/user-event';

export { customRender as render };