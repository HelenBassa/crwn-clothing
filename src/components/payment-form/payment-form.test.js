import { Elements } from "@stripe/react-stripe-js";
import { loadStripe } from "@stripe/stripe-js";

import { render } from "../../utils/test-utils";
import PaymentForm from "./payment-form.component";

const stripePromise = loadStripe("STRIPE_PUBLISHABLE_API_KEY");

it("should render PaymentForm component", () => {
  const { container } = render(
    <Elements stripe={stripePromise}>
      <PaymentForm />
    </Elements>
  );
  expect(container).toMatchSnapshot();
});
