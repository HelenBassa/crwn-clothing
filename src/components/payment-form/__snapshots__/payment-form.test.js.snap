// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PaymentForm component 1`] = `
<div>
  <div
    class="sc-gKPRtg kJaasa"
  >
    <form
      class="sc-iBYQkv iuMaOw"
    >
      <h2>
        Credit Card Payment: 
      </h2>
      <div />
      <button
        class="sc-dkrFOg sc-eDvSVe KWCcW dVmvZe sc-ftTHYK CUWzl"
      >
        Pay now
      </button>
    </form>
  </div>
</div>
`;
